packages/common/src/stream.ts

```diff
@@ -321,8 +323,12 @@ export interface StreamHandler<T extends Stream> {
    * callers to clone the input state before calling into applyCommit if they don't want the input
    * state modified.
    * @param commitData - Commit data
-   * @param context - Ceramic context
+   * @param api - Interface for reading from and writing streams to ceramic network
    * @param state - The existing state to apply the commit to.
    */
-  applyCommit(commitData: CommitData, context: Context, state?: StreamState): Promise<StreamState>
+  applyCommit(
+    commitData: CommitData,
+    api: StreamReaderWriter,
```

Comment:
I just realized that this should really be a `StreamReader`, and I think this is a great opportunity to tighten the interface here.  It's always made me nervous the fact that this handler had access to a full Ceramic API.  But since this PR is so big and changing this here requires changing it in all the stream handler packages as well, feel free to just add a TODO comment right now and then make the actual change in a follow-up PR.